pkg/utils/machine/machine.go

```diff
@@ -162,7 +162,7 @@ func NodeForMachine(ctx context.Context, c client.Client, machine *v1alpha5.Mach
 // If the providerID is not resolved for a Machine, then no Nodes will map to it
 func AllNodesForMachine(ctx context.Context, c client.Client, machine *v1alpha5.Machine) ([]*v1.Node, error) {
 	// If the providerID is defined, use that value; else, use the machine linked annotation if it's on the machine
-	providerID := lo.Ternary(machine.Status.ProviderID != "", machine.Status.ProviderID, machine.Annotations[v1alpha5.MachineLinkedAnnotationKey])
+	providerID := lo.Ternary(machine.Status.ProviderID != "", machine.Status.ProviderID, "")
```

Comment:
You can just remove this line entirely and just use `machine.Status.ProviderID` everywhere

Reply:
You also need to change this handling in `NodeForMachine`